### PR TITLE
More tests for time unit conversion

### PR DESCRIPTION
**Pull request by MichaelClerx**
See #136

### CODE REVIEW

File: test/protocols/test_unit_conversion_time_conflict.txt [hunk 1]
```diff
@@ -0,0 +1,83 @@
+documentation {
```
Comment (jonc125) [inline test/protocols/test_unit_conversion_time_conflict.txt:1]: Can we get rid of the irrelevant comments from test protocols and focus on the bits that are actually key to the test?

File: test/protocols/test_unit_conversion_time_conflict.txt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+# The model interface section defines how the protocol can modify and probe the models to which it is applied.
+# Crucially, note that all references to model variables in this section use terms from the ontology declared above (with the
+# oxmeta prefix), rather than whatever variable names are used in the model.  Protocols can only access variables that are
+# annotated with terms from a declared ontology (typically the 'oxmeta' ontology for cardiac electrophysiology, although this
+# is not required by the tools).
+
+model interface {
+    # We specify the units this protocol uses for the independent variable in the system.
```
Comment (jonc125) [inline test/protocols/test_unit_conversion_time_conflict.txt:36]: So this is the only place where we need a comment, highlighting what is tested with these two lines.

File: test/protocols/test_unit_conversion_time_conflict.txt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+tasks {
+    # This single simulation task samples model outputs every millisecond for 1 second.
+    simulation sim = timecourse {
+        range time units ms uniform 0:1:1000
+    }
+}
+
+post-processing {
```
Comment (jonc125) [inline test/protocols/test_unit_conversion_time_conflict.txt:55]: We can get rid of the post-processing entirely - it duplicates other tests and isn't needed for this one.

File: test/protocols/test_unit_conversion_time_conflict.txt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    assert sim:state_variable.SHAPE[0] == state_trans.SHAPE[1]
+    assert sim:state_variable.SHAPE[1] == state_trans.SHAPE[0]
+}
+
+# The outputs section defines what quantities (raw simulation results or post-processed values)
+# are considered outputs of the experiment and stored.
+
+outputs {
+    raw_state = sim:state_variable        "State variables (raw)"
+    state     = state_trans units various "State variables"
```
Comment (jonc125) [inline test/protocols/test_unit_conversion_time_conflict.txt:72]: Similarly `state` is the only output the tests care about here.

File: test/protocols/test_unit_conversion_time_conflict.txt [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    voltage   = sim:membrane_voltage      "Transmembrane voltage"
+    time      = sim:time                  "Time"
+}
+
+# Finally we specify the graphs to produce.
+# The first one is the default displayed by the Web Lab.
+
+plots {
```
Comment (jonc125) [inline test/protocols/test_unit_conversion_time_conflict.txt:80]: No plots needed.

File: test/protocols/test_unit_conversion_time_duplicate.txt [hunk 6]
```diff
@@ -0,0 +1,83 @@
+documentation {
```
Comment (jonc125) [inline test/protocols/test_unit_conversion_time_duplicate.txt:1]: Same comments for this and subsequent protocols as for the one above.

File: test/test_unit_conversion.py [hunk 7]
```diff
@@ -10,25 +10,63 @@
...
-    # Create protocol
-    proto = fc.Protocol(os.path.join('test', 'real', 'protocols', 'GraphState.txt'))
-
-    # Set model (generates & compiles model)
+    proto = fc.Protocol(os.path.join('test', 'protocols', 'test_unit_conversion_time_independent_var_units.txt'))
     model_name = 'difrancesco_noble_model_1985'  # has time in seconds, not milliseconds
     proto.set_model(os.path.join('test', 'real', 'models', model_name + '.cellml'))
+    proto.set_output_folder('test_unit_conversion_time_independent_var_units')
+    proto.run()
+    # Some test assertions are within the protocol itself
```
Comment (jonc125) [inline test/test_unit_conversion.py:24]: This comment will no longer be valid (and indeed the key test is in this file).

**Comment by MichaelClerx** [comment]
That's pretty much what I would have said if I'd reviewed this PR, sorry! :grimacing:

**Pull request merged**